app/src/ui/push-needs-pull/push-needs-pull-warning.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react'
+import { Dispatcher } from '../../lib/dispatcher'
+import { ButtonGroup } from '../lib/button-group'
+import { DialogFooter, DialogContent, Dialog } from '../dialog'
+import { FetchType } from '../../models/fetch'
+import { Button } from '../lib/button'
+import { Repository } from '../../models/repository'
+
+interface IPushNeedsPullWarningProps {
+  readonly dispatcher: Dispatcher
+  readonly repository: Repository
+  readonly onDismissed: () => void
+}
+
+export class PushNeedsPullWarning extends React.Component<
+  IPushNeedsPullWarningProps,
+  {}
+> {
+  public render() {
+    return (
+      <Dialog
+        title={
+          __DARWIN__ ? 'Newer Commits on Remote' : 'Newer commits on remote'
+        }
+        onDismissed={this.props.onDismissed}
+        onSubmit={this.onFetch}
+        type="warning"
+      >
+        <DialogContent>
+          <p>
+            Desktop is unable to push commits to this branch because there are
+            new commits on the remote. It needs to fetch these new commits to
+            figure out how to reconcile these changes with your local commits.
```

Comment:
I wonder if it's possible to make this a bit more beginner-friendly with a small change.

"Desktop is unable to push commits to this branch because there are commits on the remote that are not present on your local branch. It needs to fetch these new commits before pushing in order to reconcile them with your local commits."

I don't love it so feel free to suggest alternatives (or disagree).